project/src/main/java/steps/LoginSteps.java

```diff
@@ -46,4 +45,9 @@ public void loginIfRequired(UserCredentials user) {
 			loginWithValidCredentials(user);
 		}
 	}
+
+	@Step("Check is login screen is active")
```

Comment:
This step name says it will check if screen is active. But it doesn't actually check anything, it just gets screen state, and check takes place in test itself.
Please rename the step to "Get state: if login screen is active" (or smth else), or move the assert into this method and rename it to assertLoginScreenIsActive.